### PR TITLE
Add node metadata

### PR DESCRIPTION
**Pull request by cube2222**

### CODE REVIEW

File: physical/datasource.go [hunk 1]
```diff
@@ -102,3 +105,7 @@ func (dsb *DataSourceBuilder) Materialize(ctx context.Context, matCtx *Materiali
 
 	return dsb.Materializer(ctx, matCtx, dbConfig, dsb.Filter, dsb.Alias)
 }
+
+func (dsb *DataSourceBuilder) Metadata() *metadata.NodeMetadata {
+	return metadata.NewNodeMeatada(dsb.Cardinality)
```
Comment (JasiekChomiak) [inline physical/datasource.go]: Typo here - I suppose you wanted to call this "NewNodeMetadata" :P

File: physical/datasource.go [hunk 2]
```diff
@@ -64,9 +65,10 @@ type DataSourceBuilder struct {
 	Filter           Formula
 	Name             string
 	Alias            string
+	Cardinality      metadata.Cardinality
```
Comment (JasiekChomiak) [inline physical/datasource.go]: You could add a short comment to describe what the goal of this field is

File: physical/distinct.go [hunk 3]
```diff
@@ -4,20 +4,21 @@ import (
 	"context"
 
 	"github.com/cube2222/octosql/execution"
+	"github.com/cube2222/octosql/physical/metadata"
 	"github.com/pkg/errors"
 )
 
 type Distinct struct {
-	Child Node
+	Source Node
```
Comment (JasiekChomiak) [inline physical/distinct.go]: I understand that this change is purely stylistic/to improve naming?

File: physical/metadata/metadata.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+type NodeMetadata struct {
+	cardinality Cardinality
+}
+
+func (meta *NodeMetadata) Cardinality() Cardinality {
+	return meta.cardinality
+}
+
+func NewNodeMeatada(cardinality Cardinality) *NodeMetadata {
```
Comment (JasiekChomiak) [inline physical/metadata/metadata.go]: Here is the name that should be refactored to "NewNodeMetadata"

File: physical/datasource.go (hunk 2 above)
Comment (cube2222) [reply physical/datasource.go]: Yea, to make the naming consistent.

File: physical/distinct.go (hunk 3 above)
Comment (cube2222) [reply physical/distinct.go]: Yea, to make the naming consistent.

**Pull request merged**